src/stores/ItemTemplateStore.js

```diff
@@ -87,6 +88,54 @@ class ItemTemplateStore {
     return embedUrl.toString();
   }
 
+  Validate = flow(function * ({objectId}) {
+    const address = this.itemTemplates[objectId]?.metadata?.public?.asset_metadata?.nft.address;
+    let problemsList = [];
+
+    // Case 1: check for missing or invalid contract address
+    if(!(address && this.client.utils.ValidAddress(address))) {
+      // Get link to the item template's settings page
+      const link = urlJoin("item-templates", objectId, "settings");
+
+      // Add to list with the link
+      problemsList.push({
+        type: "error",
+        message: "No or invalid contract address configured",
+        link
+      });
+    }
+
+    // Case 2: check for contract addresses used by multiple templates
+    // Load list of templates via database
+    yield this.LoadItemTemplates();
+
+    for (let i=0; i<this.allItemTemplates.length; i++) {
```

Comment:
this is easier to express as `this.allItemTemplates.forEach(template => {`, then you can access template.address and template.object id without having to define them as separate variables. you could also use `for(const template of this.allItemTemplates) {`